providers/posflag/posflag.go

```diff
@@ -30,11 +32,19 @@ type Posflag struct {
 // a config file. If they are not, then the default values of the flags
 // are merged. If they do exist, the flag values are not merged but only
 // the values that have been explicitly set in the command line are merged.
-func Provider(f *pflag.FlagSet, delim string, ko *koanf.Koanf) *Posflag {
+func Provider(f *pflag.FlagSet, delim string, ko *koanf.Koanf, optFuncs ...optFunc) *Posflag {
```

Comment:
Added variadic functional options in order to allow optionally supplying the callbacks for modifying the key and value.